### PR TITLE
meta: remove unused test support functions

### PR DESCRIPTION
**Pull request by WikiRik**
## Pull Request Checklist

- [X] Have you added new tests to prevent regressions?
- [X] If a documentation update is necessary, have you opened a PR to [the documentation repository](https://github.com/sequelize/website/)? 
- [X] Did you update the typescript typings accordingly (if applicable)?
- [X] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [X] Does the name of your PR follow [our conventions](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md#6-commit-your-modifications)?

## Description Of Change

Unblocks #17038

### CODE REVIEW

File: packages/core/test/integration/query-interface.test.js [hunk 1]
```diff
@@ -21,15 +20,10 @@ describe(Support.getTestDialectTeaser('QueryInterface'), () => {
   if (dialect.supports.schemas) {
     describe('dropAllSchema', () => {
       it('should drop all schema', async function () {
-        await this.queryInterface.dropAllSchemas({
-          skip: [this.sequelize.config.database],
-        });
+        await Support.dropTestSchemas(this.sequelize);
```
Comment (ephys) [inline packages/core/test/integration/query-interface.test.js:23]: this is the only change I'm opposed to as `dropAllSchema` is what is being tested and we can't guarantee we won't change `dropTestSchemas` in the future

**Review by ephys — CHANGES_REQUESTED** [review]

File: packages/core/test/integration/query-interface.test.js (hunk 1 above)
Comment (WikiRik) [reply packages/core/test/integration/query-interface.test.js:23]: Woops, you're right. Not sure how I looked over that 😅

File: packages/core/test/integration/query-interface.test.js [hunk 2]
```diff
@@ -21,15 +20,12 @@ describe(Support.getTestDialectTeaser('QueryInterface'), () => {
   if (dialect.supports.schemas) {
     describe('dropAllSchema', () => {
       it('should drop all schema', async function () {
-        await this.queryInterface.dropAllSchemas({
-          skip: [this.sequelize.config.database],
+        await this.queryInterface.dropAllSchemas({ 
+          skip: [this.sequelize.config.database],
```
Comment (WikiRik) [inline packages/core/test/integration/query-interface.test.js:24]:
```suggestion
        await this.queryInterface.dropAllSchemas({
          skip: [this.sequelize.config.database],
```
Trying to edit from my phone but it's not going great. Maybe this helps

File: packages/core/test/integration/query-interface.test.js (hunk 1 above)
Comment (WikiRik) [reply packages/core/test/integration/query-interface.test.js]: Fixed

**Review by ephys — APPROVED** [review]

**Pull request merged**